sg_otio/track_diff.py

```diff
@@ -280,26 +314,78 @@ def __init__(self, sg, sg_project, new_track, old_track=None):
         self._sg_shot_link_field_name = None
         self._counts = defaultdict(int)
         self._total_count = 0
-
+        # We need to keep references on the tracks otherwise underlying C++ objects
+        # might be freed.
+        self._old_track = old_track
+        self._new_track = new_track
         self._diffs_by_shots = {}
+
         # Retrieve the Shot fields we need to query from SG.
         sg_shot_fields = SGShotFieldsConfig(
             self._sg, None
         ).all
 
-        # Retrieve SG Entities from the old_track
-        sg_shot_ids = []
-        prev_clip_list = []
-        if old_track:
-            sg_cut = old_track.metadata.get("sg")
+        # Retrieve the SG Entity we should use for the comparison
+        # and do some sanity check
+        new_track_sg_link = None
+        sg_cut = new_track.metadata.get("sg")
+        if sg_cut:
```

Comment:
It feels to me that this whole part could be extracted as a method `set_sg_entity` which would allow to explain how we choose it in the doc, and make the code clearer, but that's just my opinion.